win32ss/user/user32/controls/ghost.c

```diff
@@ -0,0 +1,570 @@
+/*
+ * PROJECT:         ReactOS user32.dll
+ * PURPOSE:         Ghost window
+ * FILE:            win32ss/user/user32/controls/ghost.c
+ * PROGRAMER:       Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include <user32.h>
+
+WINE_DEFAULT_DEBUG_CHANNEL(ghost);
+
+#define GHOST_TIMER_ID  0xFACEDEAD
+#define GHOST_INTERVAL  1000        // one second
+
+static const WCHAR ghostW[] = L"Ghost";
+const struct builtin_class_descr GHOST_builtin_class =
+{
+    ghostW,                     /* name */
+    0,                          /* style  */
+    GhostWndProcA,              /* procA */
+    GhostWndProcW,              /* procW */
+    0,                          /* extra */
+    IDC_WAIT,                   /* cursor */
+    (HBRUSH)(COLOR_3DFACE + 1)  /* brush */
+};
+
+static HBITMAP
+IntCreate32BppBitmap(INT cx, INT cy)
+{
+    HBITMAP hbm = NULL;
+    BITMAPINFO bi;
+    HDC hdc;
+    LPVOID pvBits;
+
+    ZeroMemory(&bi, sizeof(bi));
+    bi.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bi.bmiHeader.biWidth = cx;
+    bi.bmiHeader.biHeight = cy;
+    bi.bmiHeader.biPlanes = 1;
+    bi.bmiHeader.biBitCount = 32;
+
+    hdc = CreateCompatibleDC(NULL);
+    if (hdc)
+    {
+        hbm = CreateDIBSection(hdc, &bi, DIB_RGB_COLORS, &pvBits, NULL, 0);
+        DeleteDC(hdc);
+    }
+    return hbm;
+}
+
+static HBITMAP
+IntGetWindowBitmap(HWND hwnd, INT cx, INT cy)
+{
+    HBITMAP hbm = NULL;
+    HDC hdc, hdcMem;
+    HGDIOBJ hbmOld;
+
+    hdc = GetWindowDC(hwnd);
+    if (hdc)
+    {
+        hbm = IntCreate32BppBitmap(cx, cy);
+        if (hbm)
+        {
+            hdcMem = CreateCompatibleDC(hdc);
+            if (hdcMem)
+            {
+                hbmOld = SelectObject(hdcMem, hbm);
+                {
+                    BitBlt(hdcMem, 0, 0, cx, cy, hdc, 0, 0, SRCCOPY | CAPTUREBLT);
+                }
+                SelectObject(hdcMem, hbmOld);
+                DeleteDC(hdcMem);
+            }
+        }
+        ReleaseDC(hwnd, hdc);
+    }
+    return hbm;
+}
+
+static VOID
+IntWhiten32BppBitmap(HBITMAP hbm)
```

Comment:
I find this name a little confusing since you're actually painting it gray, not white. Also, should this always be #c0c0c0 or is COLOR_3DFACE more appropriate?

Reply:
This function does bitwise-OR. This operation makes a bitmap whiter.